Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+        If the filter string matches a device serial or IP exactly, then returns just that one device.
+        If not, it will compare the device hostname instead for a match.
+        """
+        # Exact match based on device serial number
+        if not filter_str:
+            return devices
+
+        if filter_str in devices:
+            return {
+                filter_str: devices.get(filter_str)
+            }
+
+        device: PanDevice
+        for serial, device in devices.items():
+            if device.hostname == filter_str:
+                return {
+                    serial: device
+                }
+
+    def firewalls(self, filter_string: str = None) -> Generator[Firewall, None, None]:
+        """Returns an iterable of firewalls in the topology"""
+        firewall_objects = Topology.filter_devices(self.firewall_objects, filter_string)
+        if not firewall_objects:
+            raise DemistoException("Filter string returned no devices known to this topology.")
+
+        for firewall in firewall_objects.values():
+            yield firewall
+
+    def all(self, filter_string: str = None) -> Generator[Union[Firewall, Panorama], None, None]:
+        """Returns an iterable for all devices in the topology"""
+        all_devices = {**self.firewall_objects, **self.panorama_objects}
+        all_devices = Topology.filter_devices(all_devices, filter_string)
+        # Raise if we get an empty dict back
+        if not all_devices:
+            raise DemistoException("Filter string returned no devices known to this topology.")
+
+        for device in all_devices.values():
+            yield device
+
+    def get_by_filter_str(self, filter_string: str = None) -> dict:
+        """Filters all devices and returns a dictionary of matching."""
+        all_devices = {**self.firewall_objects, **self.panorama_objects}
+        all_devices = Topology.filter_devices(all_devices, filter_string)
+        return all_devices
+
+    @classmethod
+    def build_from_string(cls, hostnames: str, username: str, password: str, api_key: str = None):
+        """
+        Splits a csv list of hostnames and builds the topology based on it.
+        This function will convert each hostname/username/password/api_key combination into a PanDevice
+        object type, add them into a new instance of `Topology`, then return it.
+        """
+        topology = cls()
+        for hostname in hostnames.split(","):
+            try:
+                if api_key:
+                    device: PanDevice = PanDevice.create_from_device(
+                        hostname=hostname,
+                        api_key=api_key
+                    )
+                else:
+                    device = PanDevice.create_from_device(
+                        hostname=hostname,
+                        api_username=username,
+                        api_password=password,
+                    )
+                # Set the timeout
+                device.timeout = 120
```

Comment:
maybe define here a class member for the timeout variable

Reply:
I added this as a globalvar - in a subsequent PR, as we introduce more commands, we should make this a parameter. Not now though, because it would incorrectly cause people to assume that timeout value is honored by the entire integration.